test/functionals/api/v2/docker_registries_controller_test.rb

```diff
@@ -0,0 +1,55 @@
+require 'test_plugin_helper'
+
+module Api
+  module V2
+    class DockerRegistriesControllerTest < ActionController::TestCase
+      test 'index returns a list of all containers' do
+        get :index, {}, set_session_user
+        assert_response :success
+        assert_template 'index'
+      end
+
+      test 'index can be filtered by name' do
+        %w(thomas clayton wolfe).each do |name|
+          FactoryGirl.create(:docker_registry, :name => name)
+        end
+        get :index, { :search => 'name = thomas' }, set_session_user
+        assert_response :success
+        assert_equal 1, assigns(:registries).length
+      end
+
+      test 'creates a new registry with valid params' do
+        docker_attrs = FactoryGirl.attributes_for(:docker_registry)
+        post :create, :docker_registry => docker_attrs
+        assert_response :success
+      end
+
+      test 'does not create a new registry with invalid params' do
+        docker_attrs = FactoryGirl.attributes_for(:docker_registry)
+        docker_attrs.delete(:name)
+        post :create, :docker_registry => docker_attrs
+        assert_response 422
+      end
+
+      test 'shows a docker registry' do
+        registry = FactoryGirl.create(:docker_registry)
+        get :show, :id => registry.id
+        assert_response :success
+      end
+
+      test 'update a docker registry' do
+        registry = FactoryGirl.create(:docker_registry)
```

Comment:
All this actions that require a registry perhaps could be put in a context with a setup block that initializes the registry.